torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -10,11 +10,24 @@ namespace fuser {
 
 namespace {
 
-Val* threadPredicate(ParallelType pt) {
-  return eq(NamedScalar::getParallelIndex(pt), new Int(0));
+Val* threadPredicate(
+    ParallelType pt,
+    const ThreadPredicateMap::SourceMapType::mapped_type& sources) {
+  if (pt == ParallelType::BIDx || pt == ParallelType::BIDy ||
+      pt == ParallelType::BIDz) {
+    TORCH_INTERNAL_ASSERT(!sources.empty(), "No predicate source found");
+    TORCH_INTERNAL_ASSERT(sources.size() == 1, "Multiple sources detected");
+    auto src = *sources.begin();
+    auto flag_name = kir::getPredicateFlagName(src);
+    return new NamedScalar(flag_name, DataType::Bool);
+  } else {
+    return eq(NamedScalar::getParallelIndex(pt), new Int(0));
+  }
 }
 
-Bool* getThreadPredicate(const ir_utils::ParallelTypeBitmap& bits) {
+Bool* getThreadPredicate(
```

Comment:
what is the difference between `threadPredicate()` and `getThreadPredicate()` ?

-> an example of why consistent naming helps: either use `foo()` or `getFoo()`, but not both (since most of the code base uses the former, I'd stick with that and never use `get` in function names)

In this case I think we avoid the confusion by using more explicit names.

Reply:
Agree. Let me refactor them.